### PR TITLE
feature(github action): add-label-when-promoted to manage backport labels

### PR DESCRIPTION
**Pull request by yaronkaikov**
Similar to Scylla-dtest adding the logic of labeling PRs when merged, and adding/removing the backport labels

### CODE REVIEW

**Comment by yaronkaikov** [comment]
@fruch Why precommit is failing?

**Comment by vponomaryov** [comment]
> @fruch Why precommit is failing?

You have lines longer than it is configured for pre-commit:

```
11:35:56  diff --git a/.github/search_commits.py b/.github/search_commits.py
11:35:56  index 4009e86f..b81d9787 100755
11:35:56  --- a/.github/search_commits.py
11:35:56  +++ b/.github/search_commits.py
11:35:56  @@ -17,8 +17,10 @@ except KeyError:
11:35:56   def parser():
11:35:56       parser = argparse.ArgumentParser()
11:35:56       parser.add_argument('--repository', type=str, default='scylladb/scylla-pkg', help='Github repository name')
11:35:56  -    parser.add_argument('--commit_before_merge', type=str, required=True, help='Git commit ID to start labeling from newest commit.')
11:35:56  -    parser.add_argument('--commit_after_merge', type=str, required=True, help='Git commit ID to end labeling at (oldest commit, exclusive).')
11:35:56  +    parser.add_argument('--commit_before_merge', type=str, required=True,
11:35:56  +                        help='Git commit ID to start labeling from newest commit.')
11:35:56  +    parser.add_argument('--commit_after_merge', type=str, required=True,
11:35:56  +                        help='Git commit ID to end labeling at (oldest commit, exclusive).')
11:35:56       parser.add_argument('--label', type=str, default='promoted-to-master', help='Label to use')
11:35:56       parser.add_argument('--ref', type=str, required=True, help='PR target branch')
11:35:56       return parser.parse_args()
```

`.pre-commit-config.yaml`: 
```
autopep8 -i -j 2 --max-line-length=120 --ignore=E226,E24,W50,W690,E402,E731
```

File: .github/workflows/add-label-when-promoted.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Check if commits are promoted
+
+on:
+  push:
+    branches:
+      - master
+      - branch-*.*
```
Comment (vponomaryov) [inline .github/workflows/add-label-when-promoted.yaml:7]: We have `manager-3.2` branch and `perf-v14`.
Shouldn't those also be supported here?

Comment (fruch) [reply .github/workflows/add-label-when-promoted.yaml:7]: should add:

- branch-perf-v*
- manager-*.*

Comment (fruch) [reply .github/workflows/add-label-when-promoted.yaml:7]: and later we should add those to mergify configuration

Comment (yaronkaikov) [reply .github/workflows/add-label-when-promoted.yaml:7]: Added

**Pull request merged**